sympy/multipledispatch/tests/test_dispatcher.py

```diff
@@ -268,3 +269,14 @@ def _(a):
         raise MDNotImplementedError()
 
     assert raises(NotImplementedError, lambda: f(1.0))
+
+@XFAIL
```

Comment:
Why the xfail?

Reply:
The code raises error normally, but the test fails. Resolving this is outside of the scope of this PR, since other tests in this file suffers same error. (You can see every other test which raises error is decorated with `XFAIL` as well.)

Reply:
That doesn't make any sense to me. Can you show me how the test fails without this line?

Reply:
Opened #20012 - these tests all pass on my machine.

Reply:
Well, there's nothing much to show. This is what happens when I remove `@XFAIL`  in `test_raise_error_on_non_class` (which was in the file before this commit)
```
Traceback (most recent call last):
  File "/home/sympy/sympy/sympy/multipledispatch/tests/test_dispatcher.py", line 95, in test_raise_error_on_non_class
    assert raises(TypeError, lambda: f.add((1,), inc))
AssertionError
```

I found out that `raises` here returns `None`, not `True`.

Reply:
The PR I link above fixes that.

Reply:
How long will yours take to be merged?

Reply:
Fixed it!